enumerable.rb

```diff
@@ -1,50 +1,150 @@
+# rubocop:disable Metrics/ModuleLength
 module Enumerable
-    def my_each
-        if block_given?
-            for x in self do
-                yield (x)
-            end
-        end
-        return self
-    end
-    
-    def my_each_with_index
-        if block_given?
-            for x in self do
-                yield (x)                
-            end
-            for y in self do
-                y.index(self[x]).to_i
-            end
-        end
-        return self
-    end
-    ['a', 'b', 'c'].my_each_with_index { |el, i| puts i }
-    def my_select
+  def my_each
+    return to_enum(:my_each) unless block_given?
 
+    each do |x|
+      yield x
     end
+    self
+  end
 
-    def my_all?
+  def my_each_with_index
+    return to_enum __method__ unless block_given?
 
+    idx = 0
+    each do |x|
+      yield(x, idx)
+      idx += 1
     end
+    self
+  end
 
-    def my_any?
+  def my_select
+    return to_enum unless block_given?
 
+    array = []
+    my_each do |x|
+      array.push(x) if yield(x) == true
     end
+    array
+  end
 
-    def my_none?
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    return true if !block_given? && arg.nil? && include?(nil) == false && include?(false) == false
+    return false unless block_given? || !arg.nil?
 
+    if block_given?
+      my_each { |x| return false if yield(x) == false }
+    elsif arg.class == Regexp
+      my_each { |x| return false if arg.match(x).nil? }
+    elsif arg.class <= Numeric || arg.class <= String
+      my_each { |x| return false if x != arg }
+    else
+      my_each { |x| return false if (x.is_a? arg) == false }
     end
+    true
+  end
 
-    def my_count
+  def my_any?(arg = nil)
+    return true if !block_given? && arg.nil? && my_each { |x| return true if x == true } && empty? == false
+    return false unless block_given? || !arg.nil?
 
+    if block_given?
+      my_each { |x| return true if yield(x) == true }
+    elsif arg.class == Regexp
+      my_each { |x| return true unless arg.match(x).nil? }
+    elsif arg.class <= Numeric || arg.class <= String
+      my_each { |x| return true if x == arg }
+    else
+      my_each { |x| return true if x.class <= arg }
     end
+    false
+  end
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[nil, false, nil, false].my_any? != [nil, false, nil, false].any? 
```